contracts/TokenLocking.sol

```diff
@@ -59,11 +66,32 @@ contract TokenLocking is TokenLockingStorage, DSMath {
   {
     require(_amount > 0, "token-locking-invalid-amount");
 
+    address clnyToken = IColony(IColonyNetwork(colonyNetwork).getMetaColony()).getToken();
+    if (_token == clnyToken) {
+      bytes32 submittedHash;
+      (submittedHash,,,,,,,,,,) = IReputationMiningCycle(IColonyNetwork(colonyNetwork).getReputationMiningCycle(true)).reputationHashSubmissions(msg.sender);
+      require(submittedHash == 0x0);
+    }
+
     userLocks[_token][msg.sender].balance = sub(userLocks[_token][msg.sender].balance, _amount);
 
     require(ERC20Extended(_token).transfer(msg.sender, _amount), "token-locking-transfer-failed");
   }
 
+  function punishStakers(address[] _users) public onlyReputationMiningCycle {
```

Comment:
It feels wrong that the `rewardStakers` is in `ColonyNetworkMining` and `punishStakers` is in the locking contract however since that logic will likely undergo refactoring I won't argue against it living here for now. Meanwhile just add a `todo` to note this context.

Reply:
Can't find a `todo` added for this https://github.com/JoinColony/colonyNetwork/pull/277/files#diff-2143b2fd4ef32ee8dc03526a77a6a429R85

Reply:
I missed that, its fixed now